proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+
+* **Publish** sets the target state to _Active_. This operation can be called from the states:
+  - _Draft_ when the integration should be published for the first time
+  - _Inactive_ when the integration has been stopped
+  - _Error_ when an integration should be re-deployed in case of an error to trigger a retry
+
+* **Suspend** sets the target state to _Inactive_. This operation should be only be possible if the `currentState` is _Active_.
+
+* **Resume** is an operation which can be used when the `currentState` is _Inactive_ and the integration should be re-activated. This sets the `targetState` to _Active_
+
+* **Undeploy** marks an integration target state as _Undeployed_. An integration should be _Inactive_ before the _Undelpoyed_ state can be reached.
+
+* **Delete** is the operation which removes an integration in an _Draft_ state.  
+
+The `syndesis-rest` backend in this case will simply validate the context (i.e. whether the operation is allowed), updates the `targetState` and the returns to `syndesis-ui`. 
+A background controller detects the state change and performs the proper actions to reach the target state. 
+
+## Versioning
+
+Each integration has a unique name within a certain context (which currently is the whole installation). 
+Multiple versions of an integration can exist. 
+A _active_ integration is immutable.
+So in order to update an active integration (i.e. an integration which is in state _Active_) a new version needs to be created.
+Changing an _active_ integration in the UI causes a new integration to be created, with the following characteristics:
+* Its cloned from the _active_ integration
+* The state of the clone is set to _Draft_
+* The `parentVersion` is set to the version of the active integration it is cloned from
+* The `version` is set to undefined.
+
+Now the user is able to change anything on this integration until it gets published. When its published the following should happen:
+
+* The `version` is set to a new unique version for this integration
+* The _active_ integration's `targetState` is set to `Undeployed`
+* As soon this state has been reached (callback), this integration's `targetState` is set to `Active`.
+
+That way it is ensured that only a single version of an integration is active. 
+
+## Data model
+
+The following data model only highlights the parts which are specific to the lifecycle model:
+
+```java
+public class Integration {
+    
+    // Unique name of the integration
+    String name;
+    
+    // Current state of the integration
+    IntegrationState currentState;
+    
+    // Target state of the integration    
+    IntegrationState targetState;
+    
+    // The version of the integration which was the origin of this integration
+    // 0 if this is a new integration
+    int parentVersion;
+    
+    // Version of this integratin
+    int version;
+  
+    // Message describing the state further (e.g. error message)
+    String stateMessage;
```

Comment:
Actually this class describes the domain model which is not necessarily that what arrives at the UI. I agree that we should strictly decouple user friendly message with technical error message. Any idea how we come from an integration error to a nice UI message ?